controllers/datadogagent/controller_reconcile_agent.go

```diff
@@ -172,6 +178,12 @@ func (r *Reconciler) reconcileV2Agent(logger logr.Logger, requiredComponents fea
 		}
 		return r.cleanupV2DaemonSet(daemonsetLogger, dda, daemonset, newStatus)
 	}
+
+	// configure FIPS
+	if dda.Spec.Global.FIPS != nil && *dda.Spec.Global.FIPS.Enabled {
+		daemonset.Spec.Template = *override.ApplyFIPSConfig(logger, podManagers, dda, resourcesManager)
```

Comment:
since this is a global config, could we move this function call to be under `ApplyGlobalSettingsNodeAgent`? (and similar for the other components.)

Reply:
I put it at the end of the reconcile so that the global, feature, and override sections don't affect the fips-proxy container by adding extra env vars, mounts, etc. to try and match what we do in the helm chart

Reply:
makes sense, thanks - i'll give it some more thought